specs/eventing/data-plane.md

```diff
@@ -1,125 +1,171 @@
+cleartext (`http`) and TLS (`https`) URLs as event delivery destinations.
+
+### HTTP Verbs
+
+In the absence of specific delivery preferences, the sender MUST initiate
+delivery of the event to the recipient using the HTTP POST verb, using either
+the structured or binary encoding of the event (sender's choice). This delivery
+SHOULD be performed using the CloudEvents HTTP Binding, version 1.0.
+
+Senders MAY probe the recipient with an [HTTP OPTIONS
+request](https://tools.ietf.org/html/rfc7231#section-4.3.7); if implemented, the
+recipent MUST indicate support for the POST verb using the [`Allow`
+header](https://tools.ietf.org/html/rfc7231#section-7.4.1). Senders which
+receive an error when probing with HTTP OPTIONS SHOULD proceed using the HTTP
+POST mechanism.
+
+### Event Acknowledgement and Repeat Delivery
+
+Event recipients MUST use the HTTP response code to indicate acceptance of an
+event. The recipient MUST NOT return a response accepting the event until it has
+handled event (processed the event or stored it in stable storage). The
+following response codes are explicitly defined; event recipients MAY also
+respond with other response codes. A response code not in this table SHOULD be
+treated as a retriable error.
+
+| Response code | Meaning                     | Retry | Delivery completed | Error |
+| ------------- | --------------------------- | ----- | ------------------ | ----- |
+| `1xx`         | (Unspecified)               | Yes\* | No\*               | Yes\* |
+| `200`         | [Event reply](#event-reply) | No    | Yes                | No    |
+| `202`         | Event accepted              | No    | Yes                | No    |
+| other `2xx`   | (Unspecified)               | Yes\* | No\*               | Yes\* |
+| other `3xx`   | (Unspecified)               | Yes\* | No\*               | Yes\* |
+| `400`         | Unparsable event            | No    | No                 | Yes   |
+| `404`         | Endpoint does not exist     | Yes   | No                 | Yes   |
+| other `4xx`   | Error                       | Yes   | No                 | Yes   |
+| other `5xx`   | Error                       | Yes   | No                 | Yes   |
+
+\* Unspecified `1xx`, `2xx`, and `3xx` response codes are **reserved for future
+extension**. Event recipients SHOULD NOT send these response codes in this spec
+version, but event senders MUST handle these response codes as errors and
+implement appropriate failure behavior.
+
+<!-- TODO: Should 3xx redirects and 401 (Unauthorized) or 403 (Forbidden) errors
+be retried? What about `405` (Method Not Allowed), 413 (Payload Too Large), 414
+(URI Too Long), 426 (Upgrade Required), 431 (Header Fields Too Large), 451
+(Unavailable for Legal Reasons)? -->
+
+Recipients MUST be able to handle duplicate delivery of events and MUST accept
+delivery of duplicate events, as the event acknowledgement could have been lost
+in return to the sender. Event recipients MUST use the [`source` and `id`
```

Comment:
Is this really a MUST? I'm trying to understand what we're after here. Are we saying that if the event recipient is wanting to detect duplicates then it MUST use these fields (and maybe some others?) for the detection logic.
The way I read this (probably incorrect) is that all recipients MUST detect dupes. I do not think we want that.

Reply:
I'm trying to define _how_ the recipient can determine that two events are, in fact, duplicates. This is pretty much just copied from the CloudEvents spec in terms of what a duplicate event is.

This is trying to say two things:

1. Recipients MUST handle duplicate events in some way.
2. Duplicate events are events what have the same `source` and `id` values.

Reply:
This is language or understanding from the cloudevents spec, id is not enough to calculate idempotent-ness for a single event.

Reply:
I switched the wording from "detect" to "determine"